shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -123,6 +126,21 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   }
 }
 
+- (BOOL)searchWeb:(NSString*)searchTerm {
+  NSString* escapedText = [searchTerm
+      stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet
+                                                             URLHostAllowedCharacterSet]];
+  NSString* searchURL = [NSString stringWithFormat:@"%@%@", searchURLPrefix, escapedText];
+
+  __block BOOL result;
+  [[UIApplication sharedApplication] openURL:[NSURL URLWithString:searchURL]
+                                     options:@{}
+                           completionHandler:^(BOOL success) {
+                             result = success;
```

Comment:
is this completion block sync or async? 

If async, then updating result here won't change what this function returns.

Reply:
Ah, it's async
But somehow the test passes 👀 should I remove this check then?

Reply:
If async, then this result always returns false. 

What is this result used for? Is it piped back to the framework? If so, you may wanna have `searchWeb` function to take a block param, rather than returning a BOOL.

Reply:
Gotcha. Can remove it if you don't need the result.